File: java/simpleIO.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class simpleIO {

    public static void readAndWriteFile(String inputPath, String outputPath) {
        try (BufferedReader reader = new BufferedReader(new FileReader(inputPath));
             BufferedWriter writer = new BufferedWriter(new FileWriter(outputPath))) {

            String line;
            while ((line = reader.readLine()) != null) {
                // Escrever cada linha no novo arquivo
                writer.write(line);
                writer.newLine();
            }

        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void main(String[] args) {
        String inputFile = "data\\biggerSample.txt";
        String outputFile = "data/outputJava.txt";

        // Medindo o tempo de execução
        long startTime = System.currentTimeMillis();
        readAndWriteFile(inputFile, outputFile);
        long endTime = System.currentTimeMillis();

        // Exibindo o tempo de execução
        double executionTime = (endTime - startTime) / 1000.0; // em segundos
        System.out.println("Tempo de execução: " + executionTime + " segundos");
    }
}

File: python/simpleIO.py
import time


def read_and_write_file(input_path, output_path):
    # Ler o conteúdo do arquivo de texto
    with open(input_path, "r", encoding="utf-8") as file:
        content = file.read()

    # Escrever o conteúdo para um novo arquivo
    with open(output_path, "w", encoding="utf-8") as file:
        file.write(content)


if __name__ == "__main__":
    input_file = "data/biggerSample.txt"
    output_file = "data/outputPython.txt"

    # Medindo o tempo de execução
    start_time = time.time()
    read_and_write_file(input_file, output_file)
    end_time = time.time()

    # Exibindo o tempo de execução
    execution_time = end_time - start_time
    print(f"Tempo de execução: {execution_time} segundos")
